certificates.c

```diff
@@ -79,53 +86,139 @@ hibacert_from_ext(struct hibacert *cert, struct hibaext *ext,
 	return HIBA_OK;
 }
 
+static int
+hibacert_load_one(struct hibacert *cert, const unsigned char *data, size_t len) {
+	int ret;
+	struct sshbuf *blob = sshbuf_new();
+	sshbuf_put(blob, data, len);
+	cert->exts = xreallocarray(cert->exts, cert->nexts + 1, sizeof(struct hibaext*));
+	cert->nexts++;
+	cert->exts[cert->nexts - 1] = hibaext_new();
+	ret = hibaext_decode(cert->exts[cert->nexts - 1], blob);
+	sshbuf_free(blob);
+	return ret;
+}
+
+static struct sshbuf*
+hibacert_maybe_inflate(struct sshbuf *extensions) {
```

Comment:
I did not see corresponding deflate support, is that not implemented?

Reply:
Done. I converted the hibaext_encode_* to support multiple extensions merging implementing the compression, merging and base64 / raw encoding.

This code is exercised by the generate_test_certificates binary to confirm HIBA extensions serialized in any format can be properly deserialized.